berkdb/lock/lock.c

```diff
@@ -3177,6 +3177,19 @@ __lock_put_internal(lt, lockp, lock, obj_ndx, need_dd, flags)
 		lockp->refcount--;
 		return (0);
 	}
+	DB_LOCKER *sh_locker = lockp->holderp;
+
+	if (is_pagelock(lockp->lockobj) && IS_WRITELOCK(lockp->mode) &&
+			F_ISSET(sh_locker, DB_LOCKER_TRACK_WRITELOCKS)) {
+		for (int i = 0; i < sh_locker->ntrackedlocks; i++) {
+			if (sh_locker->tracked_locklist[i] == lockp) {
+				sh_locker->tracked_locklist[i] = sh_locker->tracked_locklist[
```

Comment:
how about making this line a bit more legible (that way you don't even need a comment):
   last_p = h_locker->tracked_locklist[sh_locker->ntrackedlocks - 1];
   sh_locker->tracked_locklist[i] = last_ptr;